### PR TITLE
API updates for Team

### PR DESCRIPTION
**Pull request by nfagerlund**
Originally filed as https://github.com/hashicorp/terraform-website/pull/844; adds visible attribute, adds patch endpoint. 

I merged that before it was ready! This one can sit in draft mode until it's time.

### CODE REVIEW

**Comment by dylanegan** [comment]
🙌

**Comment by nfagerlund** [comment]
## Do not merge

This is paused because I'm moving all of the content around right now. I'll make sure it gets rebased onto the new situation, and I'll post here again when it's possible to merge.

File: content/source/docs/cloud/api/teams.html.md [hunk 1]
```diff
@@ -106,6 +106,7 @@ Key path                              | Type   | Default   | Description
 -----------------------               |--------|-----------|------------
 `data.type`                           | string |           | Must be `"teams"`.
 `data.attributes.name`                | string |           | The name of the team, which can only include letters, numbers, `-`, and `_`. This will be used as an identifier and must be unique in the organization.
+`data.attributes.visibility` **(beta)**  | boolean | `secret` | The visibility of the team within its organization. When `organization`, the team can be seen by every user within that organization. Valid options are `secret` and `organization`.
```
Comment (nfagerlund) [inline content/source/docs/cloud/api/teams.html.md]:
```suggestion
`data.attributes.visibility` **(beta)**  | string | `secret` | The visibility of the team within its organization. When `organization`, the team can be seen by every user within that organization. Valid options are `secret` and `organization`.
```

File: content/source/docs/cloud/api/teams.html.md [hunk 2]
```diff
@@ -223,6 +227,100 @@ $ curl \
...
+
+This PATCH endpoint requires a JSON object with the following properties as a request payload.
+
+Properties without a default value are required.
+
+Key path                              | Type   | Default   | Description
+-----------------------               |--------|-----------|------------
+`data.type`                           | string |           | Must be `"teams"`.
+`data.attributes.name`                | string |           | The name of the team, which can only include letters, numbers, `-`, and `_`. This will be used as an identifier and must be unique in the organization.
+`data.attributes.visibility` **(beta)**  | boolean | (previous value) | The visibility of the team within its organization. When `organization`, the team can be seen by every user within that organization. Valid options are `secret` and `organization`.
```
Comment (nfagerlund) [inline content/source/docs/cloud/api/teams.html.md]:
```suggestion
`data.attributes.visibility` **(beta)**  | string | (previous value) | The visibility of the team within its organization. When `organization`, the team can be seen by every user within that organization. Valid options are `secret` and `organization`.
```

**Comment by dylanegan** [comment]
Thanks @nfagerlund. We'll get to this as soon as we can!

**Pull request closed without merging**